power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -131,22 +129,17 @@ class MainModelImpl {
     using OwnedUpdateDataset = typename ModelType::OwnedUpdateDataset;
     using ComponentFlags = typename ModelType::ComponentFlags;
 
-    static constexpr Idx isolated_component{main_core::isolated_component};
-    static constexpr Idx not_connected{main_core::not_connected};
-    static constexpr Idx sequential{main_core::utils::sequential};
-
   public:
     using ImplType = ModelType;
     using Options = MainModelOptions;
-    using MathState = main_core::MathState;
     using MetaData = meta_data::MetaData;
 
     // constructor with data
     explicit MainModelImpl(double system_frequency, ConstDataset const& input_data,
                            MathSolverDispatcher const& math_solver_dispatcher, Idx pos = 0)
         : system_frequency_{system_frequency},
           meta_data_{&input_data.meta_data()},
-          math_solver_dispatcher_{&math_solver_dispatcher} {
+          solver_preparation_context_{.math_solver_dispatcher = &math_solver_dispatcher} {
```

Comment:
This construction can IMO go to `MainModel` instead of `MainModelImpl` just like `JobDispatcher`

Reply:
should be fine. this touches the topic of dependency inversion, which is part of @nitbharambe 's PR